contracts/XYZSwapRouter02.sol

```diff
@@ -291,123 +379,116 @@ contract XYZSwapRouter02 is IXYZSwapRouter02 {
             (uint256 amount0Out, uint256 amount1Out) = input == token0
                 ? (uint256(0), amountOut)
                 : (amountOut, uint256(0));
-            address to = i < path.length - 2
-                ? XYZSwapLibrary.pairFor(factory, output, path[i + 2])
-                : _to;
-            address pair = XYZSwapLibrary.pairFor(factory, input, output);
-            IXYZSwapPair(pair).swap(amount0Out, amount1Out, to, new bytes(0));
+            address to = i < path.length - 2 ? pairsPath[i + 1] : _to;
+            IXYZSwapPair(pairsPath[i]).swap(amount0Out, amount1Out, to, new bytes(0));
         }
     }
 
     function swapExactTokensForTokens(
         uint256 amountIn,
         uint256 amountOutMin,
+        address[] memory pairsPath,
         IERC20[] memory path,
         address to,
         uint256 deadline
-    ) public override payable ensure(deadline) returns (uint256[] memory amounts) {
-        amounts = XYZSwapLibrary.getAmountsOut(factory, amountIn, path);
+    ) public virtual override ensure(deadline) returns (uint256[] memory amounts) {
+        veiryPairsPathSwap(pairsPath, path);
```

Comment:
Spelling error: `veiryPairsPathSwap` -> `verifyPairsPathSwap`. Apply to all thanks

Reply:
ok